File: package/main/src/String/ensureSuffix.ts
/**
 * Ensures a string ends with the given suffix, appending it only when it is
 * not already present.
 * @param {string} string_ - Input string
 * @param {string} suffix - Suffix to ensure
 * @returns {string} String guaranteed to end with the suffix
 * @example
 * ensureSuffix("file", ".txt"); // "file.txt"
 * ensureSuffix("file.txt", ".txt"); // "file.txt"
 */
export const ensureSuffix = (string_: string, suffix: string): string =>
  string_.endsWith(suffix) ? string_ : string_ + suffix;

File: package/umt_python/src/string/ensure_suffix.py
def ensure_suffix(string_: str, suffix: str) -> str:
    """
    Ensures a string ends with the given suffix, appending it only when it is
    not already present.

    Args:
        string_: Input string.
        suffix: Suffix to ensure.

    Returns:
        String guaranteed to end with the suffix.

    Example:
        >>> ensure_suffix("file", ".txt")
        'file.txt'
        >>> ensure_suffix("file.txt", ".txt")
        'file.txt'
        >>> ensure_suffix("", "/")
        '/'
        >>> ensure_suffix("abc", "")
        'abc'
    """
    return string_ if string_.endswith(suffix) else string_ + suffix
